compute/cloud-client/src/main/java/compute/WaitForOperation.java

```diff
@@ -0,0 +1,76 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package compute;
+
+// [START compute_instances_operation_check]
+
+import com.google.cloud.compute.v1.Operation;
+import com.google.cloud.compute.v1.Operation.Status;
+import com.google.cloud.compute.v1.WaitZoneOperationRequest;
+import com.google.cloud.compute.v1.ZoneOperationsClient;
+import java.io.IOException;
+import java.net.SocketTimeoutException;
+
+public class WaitForOperation {
+
+  public static void main(String[] args) throws IOException, InterruptedException {
+    // TODO(developer): Replace these variables before running the sample.
+    // client-operation-id: Specify the operation id corresponding to a particular operation.
+    String project = "your-project-id";
+    String zone = "zone-name";
+    String operationId = "client-operation-id";
+
+    waitForOperation(project, zone, operationId);
+  }
+
+  // Waits for the specified operation to complete and returns the operation result in boolean.
+  public static void waitForOperation(String project, String zone, String operationId)
+      throws IOException, InterruptedException {
+    try {
+      // Construct the Operation instance.
+      Operation operation = Operation.newBuilder().setClientOperationId(operationId).build();
+
+      // Check if the operation hasn't been completed already.
+      if (operation.getStatus() == Status.RUNNING || operation.getStatus() == Status.PENDING) {
+        // Create a zone operations client.
+        ZoneOperationsClient zoneOperationsClient = ZoneOperationsClient.create();
+
+        // Create the wait request.
+        WaitZoneOperationRequest waitZoneOperationRequest = WaitZoneOperationRequest.newBuilder()
+            .setProject(project)
+            .setZone(zone)
+            .setOperation(String.valueOf(operation.getId())).build();
+
+        // Wait for the operation to complete; default timeout is 2 mins.
+        operation = zoneOperationsClient.wait(waitZoneOperationRequest);
+        zoneOperationsClient.close();
+
+        // Check if the operation has errors.
+        if (operation.hasError()) {
+          System.out.println("Error in executing the operation ! ! " + operation.getError());
+          return;
+        }
+      }
+      System.out.println("Operation executed successfully !");
+    } catch (SocketTimeoutException e) {
+      // Handle SocketTimeoutException which is being thrown as UnknownException.
+      // (Operation will run to completion in the background)
+      System.out.println("Operation executed successfully !");
+    }
```

Comment:
This doesn't seem right to me. `SocketTimeoutException` extends IOException - do we really need to catch it here? 

It also doesn't seem like we have any guarantee the Operation has actually executed successfully at this point, just that we're no longer waiting for the end of the operation.

Reply:
This is a temporary workaround for LRO. SocketTimeoutException is masked as UnknownException while the operation is running at the background.

Reply:
I don't understand what this means. Why is this a temporary workaround? What's causing the `SocketTimeoutException`?

Reply:
The default wait timeout is 2 mins. But often, the operation takes more time (> 2mins) to execute. Hence, it throws a SocketTimeout exception.

To account for increased operation times, we can either increase the default timeout (to something >2mins) or resort to LRO. (as a solution to the above problem).
But, both these options are currently not supported by the library.

Reply:
To clarify - the `SocketTimeoutException` is thrown when `wait` exceeds 2 minutes? Or is it thrown for other reasons?

Checking the API docs for [zoneOperation.wait](https://cloud.google.com/compute/docs/reference/rest/v1/zoneOperations/wait), it says that "it waits for no more than the default deadline (2 minutes) and then returns the current state of the operation, which might be DONE or still in progress".

So this raises to me the following concerns: 
1. The comment says "// Wait for the operation to complete; default timeout is 2 mins." - however, it doesn't seem like you can wait anything but 2 minutes? 
2. So where is `SocketTimeoutException` coming from? Shouldn't it be returning an operation object?
3. Why do we say "Operation executed successfully" if the operation hasn't actually executed? Shouldn't we use a loop to hit the endpoint until it's successful? 

@lesv FYI

Reply:
Wow - this seems wrong on so many levels.  An exception should never be the success case.  It's in the very word - Exception - something out of the ordinary.  If this is the ordinary for just a short bit, you should comment that it will change, and we should try not to publish since it's a really bad thing.

Reply:
Re-reading the thread, we are trying to say *THIS API IS NOT STABLE AND IS GOING TO CHANGE - THIS CODE IS UNSTABLE BUT IS BEING SHOWN TO ALLOW YOU TO ACCESS THE NEW API*.  As I write this, and realize that it's for the Compute API, I realize that we should just fix the API before publishing the samples.